### DISCUSSION TITLE
Add support for vanilla-extract

### DISCUSSION THREAD

**Discussion by maxsbelt** [web-infra-dev/rsbuild#2298] [Ideas]
Is there any plan to add support for vanilla extract library?
https://github.com/vanilla-extract-css/vanilla-extract

**Comment by chenjiahan** [comment]
Rspack does not support `@vanilla-extract/webpack-plugin` yet, related issue: https://github.com/web-infra-dev/rspack/issues/2845

**Reply by maxsbelt** [reply]
Thanks for the reference. Seems like it works now. I've used this config:

```
plugins: [
  pluginReact({
    reactRefreshOptions: {
      exclude: [/\.css\.ts$/],
    },
  }),
],
tools: {
  rspack: {
    plugins: [new VanillaExtractPlugin()],
  },
},
```

**Reply by chenjiahan** [reply]
Yeah, VanillaExtractPlugin has been supported by Rspack, we added documentation for this: https://github.com/web-infra-dev/rspack/pull/6499

**Reply by chenjiahan** [reply]
Example added: https://github.com/rspack-contrib/rspack-examples/pull/64

**Reply by levrik** [reply]
Are there any known issues with Vanilla Extract? HMR basically doesn't work at all (changes to CSS aren't visible) and manually reloading the page results in a crash. I have to restart Rsbuild to fix it every time.

In case this is unknown, I'll try to assemble a reproduction case.

**Reply by chenjiahan** [reply]
This is an unknown case, feel free to submit an issue for this.

**Reply by levrik** [reply]
@chenjiahan Filed a bug report at https://github.com/web-infra-dev/rsbuild/issues/5906
